avaframe/in1Data/getInput.py

```diff
@@ -754,3 +755,57 @@ def getInputPaths(avaDir):
     demFile = getDEMPath(avaDir)
 
     return demFile, relFiles, relFieldFiles
+
+
+def checkForMultiplePartsShpArea(avaDir, lineDict, modName, type=''):
+    """ check if in polygon read from shape file holes are present, if so error and save a plot to Outputs/com1DFA
+        procedure: check if polygon has several parts
+
+        Parameters
+        -----------
+        avaDir: str
+            path to avalanche directory
+        lineDict: dict
+            dictionary with info read from shape file
+            used: x, y, Start, Length, nParts, nFeatures
+        modName: str
+            name of computational module where to save error plots to Outputs subfolder
+        type: str
+            type of shp file area (release, secondary release, entrainment, resistance)
+
+        Returns
+        --------
+        error if number of parts is bigger 1
+        save plot showing all parts of each polygon feature in Outputs/modName
+
+    """
+
+    foundMultipleParts = False
+    # loop over all polygons in scenario
+    for lineFeature in range(lineDict['nFeatures']):
+        if len(lineDict['nParts'][lineFeature]) > 2:
+            # fetch number of parts for each lineFeature
+            nParts = lineDict['nParts'][lineFeature]
+            # create x, y coordinates of each lineFeature
+            xFeat = lineDict['x'][int(lineDict['Start'][lineFeature]): int(lineDict['Start'][lineFeature] + lineDict['Length'][lineFeature])]
+            yFeat = lineDict['y'][int(lineDict['Start'][lineFeature]): int(lineDict['Start'][lineFeature] + lineDict['Length'][lineFeature])]
+
+            # check for type of area
+            if type.lower() in ['entrainment', 'resistance', 'secondary release']:
+                lineFileName = lineDict['fileName'].stem
+            else:
+                lineFileName = lineDict['file'].stem
+
+            title = ('Parts of lineFeature %d of %s' % (lineFeature, lineFileName))
+            pathDict = {'pathResult': pathlib.Path(avaDir, 'Outputs', modName),
+                        'title': title, 'outFileName': ('%s feature%d_errorPlot' % (type, lineFeature))}
+            in1DataPlots.plotAreaShpError(xFeat, yFeat, nParts, pathDict)
+
+            # set flag
+            foundMultipleParts = True
+
+    # if polygon has multiple parts - error
+    if foundMultipleParts:
+        message = '%s feature has holes - check shape file: %s' % (type, lineFileName)
+        log.error(message)
+        raise AssertionError(message)
```

Comment:
What is the reasoning for making this extra step instead of raising the error right when the flag is set?

Reply:
this is so that the plots are produced first of all features of the current scenarion (shp file) that have more parts